src/java/com/zimbra/cs/nginx/NginxLookupExtension.java

```diff
@@ -910,16 +908,25 @@ private DomainExternalRouteInfo getDomainExternalRouteInfo(ILdapContext zlc, Con
             for (String host: imapServerAddrs) {
                 try {
                     Server imapServer = prov.getServerByServiceHostname(host);
+                    if (imapServer == null) {
+                        ZimbraLog.nginxlookup.warn("cannot get imap server by service hostname for %s", host);
+                        continue;
+                    }
                     imapServers.add(imapServer);
                 } catch (ServiceException e) {
-                    ZimbraLog.nginxlookup.warn("cannot get imap server %s", host);
+                    ZimbraLog.nginxlookup.warn("cannot get imap server by service hostname - %s", host);
```

Comment:
nit: are we purposely formatting this warning slightly differently (or inconsistent with) than the one on line 912?
Of course, there are other inconsistencies through the code, but not sure if we're purposely adding another.

Reply:
I did do it deliberately, so that anyone wondering where the logging happened, they can work out where - given that we don't include line numbers in our logging.